### DISCUSSION TITLE
Why doesnt Arduino Nano RP2040 port support WPA_PSK in AP mode?

### DISCUSSION THREAD

**Discussion by manpowre** [micropython/micropython#10841] [RP2040 / Pico]
Why doesnt Arduino Nano RP2040 port support WPA_PSK in AP mode ?

The reason I ask is that on android there is a security patch that came early last year (dec 2021 on pixel phones) that doesnt allow to autoconnect to WEP network anymore after you install the security patch. Have to connect manual. Now even older android tablets come pre-configured with this security patch.

Port:
MicroPython v1.19.1-891-g8d9a7fd22 on 2023-02-23; Arduino Nano RP2040 Connect with RP2040

`print(dir(network.WLAN))`
['__class__', '__name__', '__bases__', '__dict__', 'OPEN', 'WEP', 'WPA_PSK', 'active', 'config', 'connect', 'disconnect', 'ifconfig', 'ioctl', 'isconnected', 'scan', 'status']

```
import network
nic = network.WLAN(network.AP_IF)
nic.config("ssid", "pass", security=nic.WPA_PSK)
```

I get the following:
OSError: AP mode supports WEP security only

Sidenote:
On ESP32 for micropython this works fine:
```
nic.config(ssid='ssid') 
nic.config(password='pass') 
nic.config(max_clients=10) 
nic.config(authmode=network.AUTH_WPA_WPA2_PSK) 
```

**Comment by manpowre** [comment]
anyone? 
I tested Pico W, and it supports WPA2 with micropython.

**Reply by th3w** [reply]
each uses a different driver because of different wlan hardware:
Pico W: CYW43
Arduniono Nano RP2040: NINAW10

 i think, the ESP32 is using a ESP32 specific driver too.
